qiskit/providers/ibmq/utils/converters.py

```diff
@@ -42,6 +44,30 @@ def utc_to_local(utc_dt: Union[datetime.datetime, str]) -> datetime.datetime:
     return local_dt
 
 
+def datetime_to_str(date_time: datetime.datetime) -> str:
+    """Convert a datetime object to a formatted string representation.
+
+    Args:
+        date_time: Input `datetime` object.
+
+    Returns:
+        A string representation of the `datetime`.
+    """
+    return date_time.strftime(DATETIME_TO_STR_FORMATTER)
+
+
+def str_to_datetime(date_time: str) -> datetime.datetime:
+    """Convert a formatted string representing a date time to a datetime object.
+
+    Args:
+        date_time: Input string to convert into a datetime object.
+
+    Returns:
+        The datetime representing the string.
+    """
+    return datetime.datetime.strptime(date_time, DATETIME_TO_STR_FORMATTER)
```

Comment:
Why not use `dateutil.parser`? It gives you a lot more flexibility than parsing just a specific format.